apps/fft/src/fft.fil

```diff
@@ -0,0 +1,194 @@
+import "primitives/signed.fil";
+import "helpers.fil";
+import "butterfly.fil";
+import "primitives/state.fil";
+import "primitives/reshape.fil";
+
+/// Implements a combinational PEASE fft dataflow.
+/// Uses `N` butterflies
+comp PeaseFFT[NStages, N]<'G: L>(
+  go: interface['G],
+  in[NPoints][2]: ['G, 'G+1] W
+) -> (
+  out[NPoints][2]: ['G+L, 'G+L+1] W
+) with {
+  let NPoints = pow2(NStages);
+  some L where L > 0;
+  // Floating Point specification
+  let W = 32;
+  let E = 8;
+  let M = 23;
+} where
+  // number of butterflies (N) must divide the total number of butterflies per stage (NPoints/2)
+  NPoints % (2*N) == 0,
+  N > 0,
+  N * 2 <= NPoints,
+  NStages > 0
+{
+  // cvc5 stalls trying to prove this...
+  assume NPoints % 2 == 0;
+
+  assume pow2(NStages-1) == NPoints/2;
+  assume pow2(NStages+1) == NPoints*2;
+  assume pow2(NStages+2) == NPoints*4;
+
+  bitrev := new BitRev[NStages, W]<'G>(in{0..NPoints}{0..2});
+  // butterfly components
+  B := new Butterflies[N, W, E, M];
```

Comment:
Let's add borrow annotations to these so that we know that we have the right delays